### PR TITLE
[Merged by Bors] - refactor(Archive/Arithcc): avoid defEq abuse

### PR DESCRIPTION
**Pull request by nomeata**
the `rfl` was a bit of a defeq abuse: The goal was `StateEqRs a b` for
definitely different terms `a` and `b`, but it was still using
`StateEqRs.refl a`, because then after unfolding `StateEqRs` and doing a
bit of computation the proof held. This would break once
https://github.com/leanprover/lean4/pull/3714
lands, and is a bit fishy in any case so let's just let `simp` handle
it.

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Maybe it is time to remove the porting note?

bors d+

File: Archive/Arithcc.lean [hunk 1]
```diff
@@ -279,10 +279,8 @@ theorem compiler_correctness
   | const => simp [StateEq, step]; rfl
   -- 5.II
   | var =>
-    simp [hmap, StateEq, step] -- Porting note: was `finish [hmap, StateEq, step]`
-    constructor
-    · simp_all only [read, loc]
-    · rfl
+    -- Porting note: was `finish [hmap, StateEq, step]`
```
Comment (nomeata) [inline Archive/Arithcc.lean:282]:
```suggestion
```

**Comment by nomeata** [comment]
bors merge

**Pull request closed without merging**